### ISSUE TITLE
Detached part not found when copying across servers

### ISSUE THREAD

**Issue by jasonho-lynx** [ClickHouse/ClickHouse#23545]
Hi, I am trying to copy clickhouse data from `server A` to `server B` by following the instructions in https://clickhouse.tech/docs/en/sql-reference/statements/alter/partition/#alter_freeze-partition. However, after moving a single part `202010_70_80_5` of table `my_table` over to `server B` to test things out, I was not able to `ALTER TABLE my_table ATTACH PART '202010_70_80_5'` as it gave the following error: `Detached part "202010_70_80_5" not found.`

These were the steps:

1. On the clickhouse server docker container on `server A`, ran `ALTER TABLE my_table FREEZE PARTITION 202010`
2. Found the corresponding folder in `/var/lib/clickhouse/shadow/N`. The folder was named `202010_70_80_5`, and it contained some `txt`, `bin` and `mrk2` files.
3. Copied the folder over to a docker container on `server B`, placing it inside `/var/lib/clickhouse/data/<DATABASE>/my_table/detached/`. 
4. Also copied the table metadata and used it to create `my_table` on `server B`
5. Ran `ALTER TABLE my_table ATTACH PART '202010_70_80_5'`

Then the error `DB::Exception: Detached part "202001_70_80_5" not found.` showed up.

Am I missing any steps here?

**Comment by den-crane** [comment]
202010_70_80_5
202**0**01_70_80_5

Check privileges. Clickhouse should be owner of folder in ./detached

**Comment by jasonho-lynx** [comment]
Thanks for spotting the typo @den-crane, I will update the question.

Clickhouse was already set as folder owner, and I believe the error message would have been something like 'Permission denied...' if there were issues with privileges.

After a good amount of digging, I figured it out thanks to this [comment](https://github.com/ClickHouse/ClickHouse/issues/15504#issuecomment-748237603). For anyone else who encounters this:

Turns out `server B` had a more updated version of clickhouse, which doesn't store data inside `var/lib/clickhouse/data` like `server A` did. Instead, each table within `var/lib/clickhouse/data/<DATABASE>` is symlinked to `var/lib/clickhouse/store/<CHARS>/<CHARS>` (running `ll` in  `var/lib/clickhouse/data/<DATABASE>` showed the symlinks). So, for each table, I had to place the `PARTS` inside its corresponding `var/lib/clickhouse/store/<CHARS>/<CHARS>/detached` folder. Once that was done, `ALTER TABLE ATTACH PART` finally worked. Unfortunately, the `<CHARS>` appeared to be random, so moving the `PARTS` was a rather tedious process. For materialized views, it was even tougher as inside `var/lib/clickhouse/data/<DATABASE>`, they were saved as something like `%2Einner_id%2E<CHARS>` instead of the names of the views themselves. 

I guess the easier alternative would be to downgrade the clickhouse version on `server B`, and that probably would have made it work out of the box?

I'm sure there were good reasons for this `data` and `store` symlink update - hoping there'll be easier ways to migrate data on newer clickhouse versions!
